tests/Indicator/PhpUnitIndicatorTest.php

```diff
@@ -0,0 +1,266 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Indicator;
+
+use PhpCsFixer\Indicator\PhpUnitIndicator;
+use PhpCsFixer\Tokenizer\Tokens;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @author Andreas Möller <am@localheinz.com>
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\Indicator\PhpUnitIndicator
+ */
+final class PhpUnitIndicatorTest extends TestCase
+{
+    public function testIsPhpUnitClassThrowsLogicExceptionIfTokenAtIndexIsNotTClass()
+    {
+        $code = <<<'PHP'
+<?php
+
+class FooTest extends \PHPUnit_Framework_Testcase
+{
+}
+PHP;
+
+        $tokens = Tokens::fromCode($code);
+        $index = $tokens->getNextTokenOfKind(0, [
+            [
+                T_WHITESPACE,
+            ],
+        ]);
+        $token = $tokens[$index];
+
+        $indicator = new PhpUnitIndicator();
+
+        $this->expectException(\LogicException::class);
+        $this->expectExceptionMessage(sprintf(
+            'No T_CLASS at given index %d, got %s.',
+            $index,
+            $token->getName()
+        ));
+
+        $indicator->isPhpUnitClass(
+            $tokens,
+            $index
+        );
+    }
+
+    /**
+     * @dataProvider provideFailCases
+     *
+     * @param Tokens $tokens
+     * @param int    $index
+     */
+    public function testIsPhpUnitClassReturnsFalse(Tokens $tokens, $index)
+    {
+        $indicator = new PhpUnitIndicator();
+
+        $this->assertFalse($indicator->isPhpUnitClass(
+            $tokens,
+            $index
+        ));
+    }
+
+    public function provideFailCases()
+    {
+        $cases = [
+            'nothing' => '<?php
+
+class Foo
+{}
+',
+            'nothing-extends-old' => '<?php
+
+class Foo extends \PHPUnit_Framework_TestCase
```

Comment:
I am not sure about this case.

While I personally would mark an abstract test case as `abstract`, maybe there are people who don't. Then this would indicate a false negative for them (also case below).